requirements-dev.txt

```diff
@@ -2,5 +2,7 @@
 addict
 black
 flake8
+lintrunner
```

Comment:
this is needless https://github.com/microsoft/Olive/blob/main/.github/workflows/lint.yml#L45

Reply:
I added this to dev file mainly for other contributors. Since we are using lint for PR validation, they should have this package installed from requirements-dev file. And know how to use it, which means we also need to update [Contributing](https://github.com/microsoft/Olive/blob/main/CONTRIBUTING.md#contributing)

Reply:
Then we'd better add another requirement file. the dev requirement is used by others like testing, which will result the two packages are installed for all the test code. Ideally, the two packages should only be installed when needed.

Reply:
On another hand, any PR would trigger the lint CI pipelines. I guess it is only used locally. right? If yes, we can add the two lines in readme.

Reply:
We already have `requirements-test.txt` for test dependencies? This `requirements-dev.txt` is for dev purpose.

Reply:
Oh, sorry, it seems requirements-test.txt doesn't include requirements-dev.txt. My bad.

However, I am still prefer not put the two dependency into the requirements-dev.txt since besides installing them, the init is still needed. From this point of view, it would be better to add this stuff into readme.

Reply:
Another example: https://github.com/microsoft/onnxruntime/blob/main/requirements-dev.txt

Reply:
I think `requirements-dev.txt` in onnxruntime repo has different meaning. This file in onnxruntime repo means dev env for onnxruntime package in order to build onnxruntime from source, like [this](https://github.com/microsoft/onnxruntime/blob/main/onnxruntime/python/tools/transformers/models/stable_diffusion/README.md?plain=1#L102). Since Olive doesn't need this feature, the packages now listed in our `requirements-dev.txt` are more like dev tools for contributers, like `black`, `flake8` or `pre-commit`. I agree we should update documentation about how to use lint, but I prefer those are documented in Contributing, like [this](https://github.com/microsoft/onnxruntime/blob/main/docs/Coding_Conventions_and_Standards.md?plain=1#L153).

Also I believe most of the packages in our `requirements-dev.txt` are not needed anymore. We can remove others but only keep lint packages.

Reply:
could the pre-commit https://github.com/microsoft/Olive/pull/607/files#diff-63a9c44a44acf85fea213a857769990937107cf072831e1a26808cfde9d096b9 accomplish these things?

Reply:
Originally, the purpose of requirements-dev.txt is same with that one in ORT. The typical user scenario is:
* user git clone code
* pip install -r requirements-dev.txt.

Reply:
If you add the linter runner here, please remember remove those in the workflow.yaml

Reply:
Do we need both pre-commit and lint? I thought we will use lint to replace pre-commit?

My expectations for each requirements files in Olive are:
- `requirements.txt`: Olive dependencies.
- `requirements-test.txt`: Olive tests dependencies to run tests.
- `requirements-dev.txt`: for contributors to help them to raise a PR. This should fully support necessary packages for each PR validation, which means this should include `requirements-test.txt`.

Reply:
I can clean up the requirements files and remove unneeded packages.

Reply:
I will also update the contributing doc.

Reply:
@trajepl tried to remove the pre-commit last week. But failed. So from the time being, we need both. Jiapeng would have more context.

Reply:
Please help add one section like https://github.com/microsoft/Olive#formatting for enabling linter

Reply:
> Do we need both pre-commit and lint? I thought we will use lint to replace pre-commit?
> 
> My expectations for each requirements files in Olive are:
> 
> * `requirements.txt`: Olive dependencies.
> * `requirements-test.txt`: Olive tests dependencies to run tests.
> * `requirements-dev.txt`: for contributors to help them to raise a PR. This should fully support necessary packages for each PR validation, which means this should include `requirements-test.txt`.

Please make requirements-dev.txt doesn't include requirements-test.txt. Basically, requirements-dev.txt is mainly for develop purpose like run flake, black format, linter etc. The test is not mandatory since it is optional run